docs/website/docs/developers/general/contributing.md

```diff
@@ -2,21 +2,160 @@
+to track such work first. Pure style changes are unlikely to be accepted unless
+they are applied consistently across the project.
+
+??? tip - "Tip - code formatters and lint scripts"
+
+    Formatters like
+    [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) (C/C++) and
+    [_Black_](https://black.readthedocs.io/en/stable/) (Python) can be set to
+    run automatically in your editor of choice.
+
+    The script at
+    [`build_tools/scripts/lint.sh`](https://github.com/openxla/iree/blob/main/build_tools/scripts/lint.sh)
+    can also be used to run the full suite of lint checks.
+
+### :octicons-code-review-16: Code reviews
+
+All submissions, including submissions by maintainers, require review. We
+use GitHub pull requests (PRs) for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
+information on using pull requests.
+
+* Please keep PRs small (focused on a single issue) to make reviews and later
+  culprit-finding easier.
+* You may see trusted core contributors bending this rule for project
+  maintenance and major subsystem renovation. If you feel like the rules aren't
+  working for a certain situation, please ask as we bias towards pragmatism for
+  cases that require it.
+
+### :material-check-all: GitHub Actions workflows
+
+We use [GitHub Actions](https://docs.github.com/en/actions) to automatically
+build and test various parts of the project.
+
+* Most presubmit workflows will only run automatically on PRs if you are a
+  project collaborator. Otherwise a maintainer must
+  [approve workflow runs](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks).
+  If you are sending code changes to the project, please ask to be added as a
+  collaborator, so that these can run automatically.
+* It is generally expected that PRs will only be merged when all checks are
+  passing. In some cases, pre-existing failures may be bypassed by a maintainer.
+
+??? tip - "Tip - adjusting workflow behavior"
+
+    Some workflows only run on commits after they are merged. See the
+    [CI behavior manipulation](#ci-behavior-manipulation) section below to
+    learn how to customize this behavior.
+
+<!-- TODO(scotttodd): link to infrastructure / CI docs when they exist -->
+<!-- TODO(scotttodd): link to "obtaining commit access" -->
+
+### :octicons-git-pull-request-16: Merging approved changes
+
+After review and presubmit checks, PRs should typically be merged using
+"squash and merge".
+
+* The squashed commit summary should match the PR title and the commit
+  description should match the PR body (this is the default behavior).
+  Accordingly, please write these as you would a helpful commit message.
+
+It is assumed that the PR author will merge their change unless they ask
+someone else to merge it for them (e.g. because they don't have write access
+yet).
+
+### :octicons-git-merge-16: Obtaining commit access
```

Comment:
Wrote this section :)

Live preview: https://scotttodd.github.io/iree/developers/general/contributing/#obtaining-commit-access

![image](https://github.com/openxla/iree/assets/4010439/bdede856-e084-403b-aaa2-7049479e6c19)

The form goes to Jacques to review access requests